tests/unit/amo/reducers/test_addonsByAuthors.js

```diff
@@ -0,0 +1,129 @@
+        // We slightly modify the state to be sure that an unrelated action
+        // does not change it.
+        byAddon: {
+          'addon-slug': {},
+        },
+      };
+      const newState = reducer(state, { type: 'UNRELATED' });
+      expect(newState).toEqual(state);
+    });
+
+    describe('LOAD_OTHER_ADDONS_BY_AUTHORS', () => {
+      it('allows an empty list of add-ons', () => {
+        const state = reducer(undefined, loadOtherAddonsByAuthors({
+          slug: 'addon-slug',
+          addons: [],
+        }));
+        expect(state.byAddon).toEqual({
+          'addon-slug': [],
+        });
+      });
+
+      it('adds related add-ons by slug', () => {
+        const state = reducer(undefined, loadOtherAddonsByAuthors({
+          slug: 'addon-slug',
+          addons: [fakeAddon],
+        }));
+        expect(state.byAddon).toEqual({
+          'addon-slug': [fakeAddon],
+        });
+      });
+
+      it('excludes the main add-on from the list of add-ons', () => {
+        const state = reducer(undefined, loadOtherAddonsByAuthors({
+          slug: fakeAddon.slug,
+          addons: [fakeAddon],
+        }));
+        expect(state.byAddon).toEqual({
+          [fakeAddon.slug]: [],
+        });
+      });
+
+      it('adds OTHERS_ADDONS_BY_AUTHORS_TO_LOAD to the state', () => {
+        const slug = 'addon-slug';
+        const state = reducer(undefined, loadOtherAddonsByAuthors({
+          slug,
+          // This is the case where there are more add-ons loaded than needed.
+          addons: Array(OTHERS_ADDONS_BY_AUTHORS_TO_LOAD + 2).fill(fakeAddon),
+        }));
+        expect(state.byAddon[slug])
+          .toHaveLength(OTHERS_ADDONS_BY_AUTHORS_TO_LOAD);
+      });
+    });
+  });
+
+  describe('fetchOtherAddonsByAuthors()', () => {
+    it('throws an error if arguments are missing', () => {
+      expect(() => {
+        fetchOtherAddonsByAuthors({});
+      }).toThrow(/errorHandlerId is required/);
+      expect(() => {
+        fetchOtherAddonsByAuthors({ errorHandlerId: 'id' });
+      }).toThrow(/An add-on slug is required/);
+      expect(() => {
+        fetchOtherAddonsByAuthors({ errorHandlerId: 'id', slug: 'addon-slug' });
+      }).toThrow(/An add-on type is required/);
+      expect(() => {
+        fetchOtherAddonsByAuthors({
+          errorHandlerId: 'id',
+          slug: 'addon-slug',
+          type: ADDON_TYPE_THEME,
+          addonType: ADDON_TYPE_THEME,
+        });
+      }).toThrow(/Authors are required/);
+    });
+
+    it('returns an array of authors in payload', () => {
+      const authors = ['john', 'snow'];
+      const { payload } = fetchOtherAddonsByAuthors({
+        errorHandlerId: 'any-error-handler-id',
+        slug: 'addon-slug',
+        authors,
+        addonType: ADDON_TYPE_THEME,
+      });
+      expect(payload.authors).toEqual(authors);
+    });
+
+    it('transforms a single author into an array of authors', () => {
+      const authors = 'john';
```

Comment:
Ah, this answers my question for why you needed `concat()`. I think we can live without this shortcut since it's not overly verbose to type `authors = ['john']`. This kind of optimization is typical in JavaScript but it's only good for introducing bugs IMHO :/

Reply:
Ok, sounds good to me